### PR TITLE
Get distance between coordinates for suggestions

### PR DESCRIPTION
**Pull request by emaohn**
- getDistance: function that takes two coordinates and a travel mode, returns distance and duration of travel
- When clicking between two travel objects, we get a radius based on the distance between the two points so that we get suggestions within the two.

### CODE REVIEW

File: step-capstone/src/components/Pages/Trip.js [hunk 1]
```diff
@@ -261,17 +262,18 @@ export default class Trip extends React.Component {
         todayEndTime.setHours(23, 59, 59);
 
         if (this.state.map) {
+            console.log(this.state.selectedTimeslot && this.state.selectedTimeslot.radius ? this.state.selectedTimeslot.radius : this.state.tripSetting.userPref.radius)
```
Comment (hieudan225) [inline step-capstone/src/components/Pages/Trip.js]: you may want to remove this

File: step-capstone/src/components/Sidebars/TimeLine.js [hunk 2]
```diff
@@ -37,6 +37,10 @@ export default class TimeLine extends React.Component {
   componentDidUpdate(prevProps) {
     if (prevProps.displayDate.toDateString() !== this.props.displayDate.toDateString) {
```
Comment (hieudan225) [inline step-capstone/src/components/Sidebars/TimeLine.js:38]: should it be:
if (prevProps.displayDate.toDateString() !== this.props.displayDate.toDateString())?

File: step-capstone/src/components/Sidebars/InsertObjectPopover.js [hunk 3]
```diff
@@ -12,29 +13,46 @@ export default function InsertObjectPopover(props) {
...
-    }
+    })
+  }
 
-    return coordinates;
+  const calculateRadius = (start, end) => {
+    return new Promise(res => {
+      getDistance(start, end, "DRIVING").then((result) => {
+        let distance = result.distance.value;
+        // radius is half distance between two points + additional 2 km for encompassing start and endpoints.
```
Comment (hieudan225) [inline step-capstone/src/components/Sidebars/InsertObjectPopover.js:51]: you may want to modify the comments since 2km is no longer relevant, isn't it?

Comment (zirving) [inline step-capstone/src/components/Sidebars/InsertObjectPopover.js:51]: still have 2 km in the comment here even though gone from code

**Pull request merged**